### PR TITLE
provide configuration for rules ignoring

### PR DESCRIPTION
**Pull request by hd-deman**
related issue - https://github.com/AtomLinter/linter-csslint/issues/15

### CODE REVIEW

File: lib/init.coffee [hunk 1]
```diff
@@ -1,8 +1,24 @@
 path = require 'path'
 
 module.exports =
-  configDefaults:
-    csslintExecutablePath: path.join __dirname, '..', 'node_modules', '.bin'
+  config:
+    executablePath:
+      title: 'Csslint executable path'
```
Comment (dmnd) [inline lib/init.coffee]: nit: On the web site they call it "CSS Lint"

File: lib/init.coffee [hunk 2]
```diff
@@ -1,8 +1,24 @@
...
+      description: 'This option allows you to specify which rules to turnoff. The rules are represented as a comma-delimited list of rule IDs.'
+      type: 'array'
+      default: []
+      items:
+        type: 'string'
 
   activate: ->
+    # Internal: Prevent old deprecated config to be visible in the package settings
+    # Keep the old config settings
+    if (atom.config.get('linter-csslint.cssLintExecutablePath'))
```
Comment (dmnd) [inline lib/init.coffee]: nice

File: lib/linter-csslint.coffee [hunk 3]
```diff
@@ -27,10 +27,22 @@ class LinterCsslint extends Linter
   constructor: (editor)->
     super(editor)
 
-    atom.config.observe 'linter-csslint.csslintExecutablePath', =>
-      @executablePath = atom.config.get 'linter-csslint.csslintExecutablePath'
+    atom.config.observe 'linter-csslint.executablePath', =>
+      @executablePath = atom.config.get 'linter-csslint.executablePath'
+
+    atom.config.observe 'linter-csslint.ignore', =>
```
Comment (dmnd) [inline lib/linter-csslint.coffee]: This should be `linter-csslint.ignoreRules`

File: lib/linter-csslint.coffee [hunk 4]
```diff
@@ -9,7 +9,7 @@ class LinterCsslint extends Linter
 
   # A string, list, tuple or callable that returns a string, list or tuple,
   # containing the command line (with arguments) used to lint.
-  cmd: 'csslint --format=compact'
+  cmd: ['csslint --format=compact']
```
Comment (dmnd) [inline lib/linter-csslint.coffee]: I don't think this initial value will be use anymore, will it? The call to `observe` below will initialize it.

**Comment by dmnd** [comment]
looks mostly good to me.

It's a shame csslint doesn't have configuration in a file. Storing config in linter config makes multiple projects annoying.

**Comment by hd-deman** [comment]
@dmnd Thanks. Checking each other's code is very useful.
And :tada: :balloon:  happy birthday to you ::) :tada: :balloon:

**Comment by dmnd** [comment]
haha thanks @hd-deman :grinning:

**Comment by steelbrain** [comment]
Closing as abandoned.

**Pull request closed without merging**